src/main/kotlin/com/github/jengelman/gradle/plugins/shadow/ShadowKmpAppPlugin.kt

```diff
@@ -0,0 +1,112 @@
+package com.github.jengelman.gradle.plugins.shadow
+
+import com.github.jengelman.gradle.plugins.shadow.ShadowBasePlugin.Companion.shadow
+import com.github.jengelman.gradle.plugins.shadow.ShadowJavaPlugin.Companion.shadowJar
+import com.github.jengelman.gradle.plugins.shadow.internal.applicationExtension
+import com.github.jengelman.gradle.plugins.shadow.internal.distributions
+import com.github.jengelman.gradle.plugins.shadow.internal.javaPluginExtension
+import com.github.jengelman.gradle.plugins.shadow.internal.javaToolchainService
+import com.github.jengelman.gradle.plugins.shadow.internal.requireResourceAsText
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.plugins.ApplicationPlugin
+import org.gradle.api.tasks.JavaExec
+import org.gradle.api.tasks.Sync
+import org.gradle.api.tasks.TaskContainer
+import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.application.CreateStartScripts
+import org.gradle.jvm.application.scripts.TemplateBasedScriptGenerator
+import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
+import org.jetbrains.kotlin.gradle.targets.jvm.KotlinJvmBinaryDsl
+
+/**
+ * A [Plugin] which packages and runs a project as a Java Application using the shadowed jar.
+ *
+ * Modified from [org.gradle.api.plugins.ApplicationPlugin.java](https://github.com/gradle/gradle/blob/45a20d82b623786d19b50185e595adf3d7b196b2/platforms/jvm/plugins-application/src/main/java/org/gradle/api/plugins/ApplicationPlugin.java).
+ */
+public abstract class ShadowKmpAppPlugin : Plugin<Project> {
+  override fun apply(project: Project) {
+    project.extensions.getByType(KotlinMultiplatformExtension::class.java).jvm().binaries {
+      executable {
+        project.addRunTask(this)
+      }
```

Comment:
Can't be configured twice.

https://github.com/JetBrains/kotlin/blob/280b570e1f884f0127c5079e11c412870faa2686/libraries/tools/kotlin-gradle-plugin/src/common/kotlin/org/jetbrains/kotlin/gradle/targets/jvm/KotlinJvmBinariesDsl.kt#L102-L104

Reply:
Could do something like #1378.